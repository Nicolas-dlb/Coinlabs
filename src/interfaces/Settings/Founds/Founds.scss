.settings_founds {
  width: 86%;
  align-self: flex-end;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  padding: 20px;
  border-radius: 15px;
  background: var(--third-bg-color);
  margin-top: 2vh;
  transition: background-color 0.3s ease;
  .settings_founds_input {
    display: flex;
    align-items: center;
    width: 100%;
    button {
      margin-top: 13px;
      margin-left: 15px;
    }
  }
  .input {
    display: flex;
    margin-top: 2vh;
    border: 1px solid var(--third-font-color);
    border-radius: 10px;

    width: 60%;
    align-items: center;
    p {
      margin-left: 10px;
      font-size: 19px;
      color: var(--primary-font-color);
    }
    input {
      background: var(--third-bg-color);
      border: none;
      outline: none;
      transition: background-color 0.3s ease;
      color: var(--primary-font-color);
      caret-color: #1baa63;
      border-radius: 10px;
      font-family: inherit;
      padding: 8px;
      width: 100%;
      &::-webkit-outer-spin-button,
      &::-webkit-inner-spin-button {
        -webkit-appearance: none;
        margin: 0;
      }

      &[type="number"] {
        -moz-appearance: textfield;
      }
      &::placeholder {
        font-family: inherit;
        color: #484558;
        font-size: 12px;
      }
    }
  }
}
