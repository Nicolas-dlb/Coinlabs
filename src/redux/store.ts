import { configureStore } from "@reduxjs/toolkit";
import marketReducer from "./reducers/marketSlice";
import walletsReducer from "./reducers/walletsSlice";
import appReducer from "./reducers/appSlice";
import userReducer from "./reducers/userSlice";

export default configureStore({
  reducer: {
    market: marketReducer,
    wallets: walletsReducer,
    app: appReducer,
    user: userReducer,
  },
});
