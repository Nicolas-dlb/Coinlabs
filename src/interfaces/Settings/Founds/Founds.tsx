/* eslint-disable no-unused-vars */
import React, { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import "./Founds.scss";
import $ from "jquery";
import { selectWallets, setWallets } from "redux/reducers/walletsSlice";

function Founds() {
  const dispatch = useDispatch();
  const wallet = useSelector(selectWallets);
  const [amount, setAmount] = useState(0);
  const setWallet = () => {
    dispatch(setWallets({ usd: wallet.usd + amount }));
    $("#add_founds").val("");
    setAmount(0);
  };
  return (
    <div className="settings_founds">
      <h3>Founds</h3>
      <div className="settings_founds_input">
        <div className="input">
          <p>$</p>
          <input
            id="add_founds"
            onChange={(e) => setAmount(Number(e.target.value))}
            type="number"
            min="0"
            placeholder="Enter amount"
            onKeyPress={(e) => {
              if (e.key === "Enter") {
                setWallet();
              }
            }}
          />
        </div>
        <button
          className="btn_settings"
          onClick={() => {
            setWallet();
          }}
          type="button"
        >
          Add
        </button>
      </div>
    </div>
  );
}

export default Founds;
