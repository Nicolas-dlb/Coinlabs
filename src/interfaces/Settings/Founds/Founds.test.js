import React from "react";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import store from "redux/store";
import { Provider } from "react-redux";
import Founds from "./Founds";

describe("Founds", () => {
  let FoundsComponent;
  beforeEach(async () => {
    await act(async () => {
      FoundsComponent = await mount(
        <Provider store={store}>
          <Founds />
        </Provider>
      );
    });
  });
  it("should render correctly", () => {
    expect(FoundsComponent).toMatchSnapshot();
  });
  it("should render Founds title", () => {
    expect(FoundsComponent.find("h3").text()).toBe("Founds");
  });
  it("should render an icon", () => {
    expect(FoundsComponent.find("p").text()).toBe("$");
  });

  it("should render an input", () => {
    expect(FoundsComponent.find("input").length).toBe(1);
  });
  it("should render a button called Add", () => {
    expect(FoundsComponent.find("button").first().text()).toBe("Add");
  });
});
