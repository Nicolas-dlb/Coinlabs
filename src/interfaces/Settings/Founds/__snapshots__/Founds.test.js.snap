// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Founds should render correctly 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Founds>
    <div
      className="settings_founds"
    >
      <h3>
        Founds
      </h3>
      <div
        className="settings_founds_input"
      >
        <div
          className="input"
        >
          <p>
            $
          </p>
          <input
            id="add_founds"
            min="0"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Enter amount"
            type="number"
          />
        </div>
        <button
          className="btn_settings"
          onClick={[Function]}
          type="button"
        >
          Add
        </button>
      </div>
    </div>
  </Founds>
</Provider>
`;
